object/user_util.go

```diff
@@ -80,6 +82,30 @@ func GetUserByFields(organization string, field string) (*User, error) {
 	return nil, nil
 }
 
+func IsPasswordSame(user *User, password string) (bool, error) {
+	organization, err := GetOrganizationByUser(user)
+	if err != nil {
+		return false, err
+	}
+
+	credManager := cred.GetCredManager(organization.PasswordType)
+
+	return credManager.IsPasswordCorrect(password, user.Password, user.PasswordSalt), nil
+}
+
+func CheckPasswordSame(user *User, password string, lang string) string {
+	isSame, err := IsPasswordSame(user, password)
+	if err != nil {
+		return err.Error()
+	}
+
+	if isSame {
+		return i18n.Translate(lang, "check:password must not be same as previous")
```

Comment:
despite the fact that this corresponds to the existing approach in the project, we can think about transferring localization to the controller level, but so far lgtm